# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/ClientWebGLContext.cpp
# Commit: 4cda148ef28f
# Full Hash: 4cda148ef28f72a0ca30b68919332996e667eec6
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-15 21:48:38
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Create a separate WebGLContext::PresentInto function for WebXR. r=jgilbert
#   
#   - Do not apply mIsCanvasDirty bailout when presenting from a opaque FBO
#   back buffer.
#   - Create a separate WebGLContext::PresentInto with more specific XR
# ==============================================================================

diff -r 0b191b707750 -r 4cda148ef28f dom/canvas/ClientWebGLContext.cpp
--- a/dom/canvas/ClientWebGLContext.cpp	Mon Jun 15 18:25:58 2020 +0000
+++ b/dom/canvas/ClientWebGLContext.cpp	Mon Jun 15 18:26:00 2020 +0000
@@ -419,7 +419,7 @@
 
 void ClientWebGLContext::Present(WebGLFramebufferJS* const fb,
                                  const layers::TextureType type) {
-  if (!mIsCanvasDirty) return;
+  if (!mIsCanvasDirty && !fb) return;
   mIsCanvasDirty = false;
   mFrontBufferSnapshot = nullptr;
 